File: BackEnd/Node.js/Wasail/models/vendor.js
'use strict';

module.exports = (sequelize, Datatypes)=>{
    return sequelize.define('vendor',{
        vendor_id:{
            type: Datatypes.INTEGER,
            primaryKey: true,
            autoIncrement: true
        },
        vendor_name:{
            type: Datatypes.STRING,
            isAlpha: true,
            notNull: true,
        },

        delivery_locations:{
            type: Datatypes.ENUM('Dha', 'Gulberg', 'State Life')
        },

        image:{
            type: Datatypes.STRING,
        },

        user_table_user_id: {
            type: Datatypes.INTEGER,
            references: {
                model: 'user_tables',
                key: 'user_id'
            },
            allowNull: false
        },
    },
        {
            underscored:true
        }
    )
}

File: FrontEnd/AppDevelopment/GroceryStoreApp/store/lib/Classes/vendor.dart
class Vendor {
  final int vendorId;
  final String vendorName;
  final String image;
  final String deliveryLocations;
  final int userId;

  Vendor({
    required this.vendorId,
    required this.vendorName,
    required this.image,
    required this.deliveryLocations,
    required this.userId
  });

  factory Vendor.fromJson(Map<String, dynamic> json) {
    return Vendor(
      vendorId: json['vendor_id'],
      vendorName: json['vendor_name'],
      image: json['image'],
      deliveryLocations: json['delivery_locations'],
      userId: json['user_table_user_id'],
    );
  }
}